File: java/src/televentas/ServicioLogistica.java
package televentas;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class ServicioLogistica {

    private final ArrayList<EmpresaTransporte> empresas = new ArrayList<>();

    public ServicioLogistica() {
    }

    public ServicioLogistica(List<EmpresaTransporte> empresasIniciales) {
        this.empresas.addAll(empresasIniciales);
    }

    public List<EmpresaTransporte> getEmpresas() {
        return Collections.unmodifiableList(new ArrayList<>(empresas));
    }

    public EmpresaTransporte seleccionarEmpresa(Pedido pedido) {
        if (empresas.isEmpty()) {
            throw new IllegalStateException("No hay empresas de transporte disponibles.");
        }
        return empresas.get(0);
    }

    public Envio generarEnvio(Pedido pedido) {
        Envio envio = new Envio(
                pedido.getIdPedido(),
                pedido.getOrden().getCliente().getDireccionEnvio(),
                null,
                "pendiente"
        );
        EmpresaTransporte empresa = seleccionarEmpresa(pedido);
        empresa.asignarEnvio(envio);
        return envio;
    }
}

File: javascript/src/televentas/servicioLogistica.js
import { Envio } from "./envio.js";

export class ServicioLogistica {
  /**
   * @param {import('./empresaTransporte.js').EmpresaTransporte[] | undefined} empresasIniciales
   */
  constructor(empresasIniciales) {
    /** @type {import('./empresaTransporte.js').EmpresaTransporte[]} */
    this.empresas = empresasIniciales != null ? [...empresasIniciales] : [];
  }

  getEmpresas() {
    return [...this.empresas];
  }

  /** @param {import('./pedido.js').Pedido} pedido */
  seleccionarEmpresa(_pedido) {
    if (this.empresas.length === 0) {
      throw new Error("No hay empresas de transporte disponibles.");
    }
    return this.empresas[0];
  }

  /** @param {import('./pedido.js').Pedido} pedido */
  generarEnvio(pedido) {
    const envio = new Envio(
      pedido.getIdPedido(),
      pedido.getOrden().getCliente().getDireccionEnvio(),
      null,
      "pendiente"
    );
    const empresa = this.seleccionarEmpresa(pedido);
    empresa.asignarEnvio(envio);
    return envio;
  }
}
